python/tvm/relay/frontend/tensorflow2.py

```diff
@@ -0,0 +1,686 @@
+                DataType type = 6;           // "type"
+                TensorShapeProto shape = 7;  // "shape"
+                TensorProto tensor = 8;      // "tensor"
+                ListValue list = 1;          // any "list(...)"            }
+        }
+    Parameters
+    ----------
+    buf: attrvalue protobuf.  <class 'tensorflow.core.framework.attr_value_pb2.AttrValue'>
+    Returns
+    -------
+    The value of the attr, as a Python object.
+    """
+    fields = ["s", "i", "f", "b", "type", "shape", "tensor", "func"]
+
+    x = buf
+
+    ret = []
+
+    if not x.WhichOneof("value"):
+        return ret
+
+    if x.HasField("list"):
+        for f in fields:
+            if getattr(x.list, f):
+                if f == "type":
+                    ret += [dtypes.as_dtype(x) for x in list(getattr(x.list, f))]
+                else:
+                    ret += list(getattr(x.list, f))
+    else:
+        for f in fields:
+            if x.HasField(f):
+                if f == "type":
+                    ret = dtypes.as_dtype(getattr(x, f))
+                else:
+                    ret = getattr(x, f)
+    return ret
+
+
+def parse_attr(attr_proto):
+    """Convert node attributes (a serialized map of key-value pairs) in a node to a dict
+    Parameters
+    ----------
+    attr_proto: <class 'google.protobuf.pyext._message.MessageMapContainer'>
+    attributes of a tf node
+    protobuf message format:
+        // tensorflow/core/framework/node_def.proto
+        message NodeDef {
+            map<string, AttrValue> attr = 5;
+        }
+    Returns
+    -------
+    Dict {string: python object}
+    Examples
+    --------
+    attributes in following node converted to {'_user_specified_name': b'x', 'dtype': tf.float32 }
+        node {
+        name: "x"
+        op: "Placeholder"
+        attr {
+            key: "_user_specified_name"
+            value {
+            s: "x"
+            }
+        }
+        attr {
+            key: "dtype"
+            value {
+            type: DT_FLOAT
+            }
+        }
+    """
+    attrs = {}
+    for key, value in attr_proto.items():
+        attrs[key] = get_attr(value)
+
+    return attrs
+
+
+def convert_place_holder(shape, node, in_type=None):
+    """convert tf place holder into relay var.
+
+    Examples
+    --------
+    a tf place holder with name "x" is converted to [Var(x, ty=TensorType([], float32))]
+    """
+
+    if shape and node.name in shape:
+        input_shape = list(shape[node.name])
+    else:
+        input_shape = tensor_util.TensorShapeProtoToList(node.attr["shape"].shape)
+        for idx, dim in enumerate(input_shape):
+            if dim < 0:
+                input_shape[idx] = Any()
+    attr = parse_attr(node.attr)
+    if in_type is not None:
+        sym = [_expr.var(node.name, type_annotation=in_type)]
+    else:
+        sym = [_expr.var(node.name, shape=input_shape, dtype=attr["dtype"].name)]
+    return input_shape, sym
+
```

Comment:
The utility functions on top are imported from `tensorflow.py`. Since they were methods inside class they had to extracted. Future refactor PR can address this to allow better code reuse. However all TF1 unit tests and models need to be tested alongside for such a refactor.